packages/backend/server/src/core/workspaces/resolvers/service.ts

```diff
@@ -125,18 +128,39 @@ export class WorkspaceService {
     return true;
   }
 
-  async sendInviteEmail(inviteId: string) {
-    const target = await this.getInviteeEmailTarget(inviteId);
+  async sendInvitationNotification(
+    inviterId: string,
+    inviteId: string,
+    sendMail: boolean
+  ) {
+    await this.queue.add('notification.sendInvitation', {
+      inviterId,
+      inviteId,
+    });
+
+    if (!sendMail) {
+      return;
+    }
 
+    const target = await this.getInviteeEmailTarget(inviteId);
     if (!target) {
       return;
     }
 
-    const owner = await this.models.workspaceUser.getOwner(target.workspace.id);
+    const setting = await this.models.userSetting.get(target.inviteeUserId);
+    // TODO(@fengmk2): should remove sendMail flag
+    if (!setting.receiveNotificationEmail) {
+      return;
+    }
+
+    const inviter = await this.models.user.getWorkspaceUser(inviterId);
+    if (!inviter) {
+      return;
+    }
 
     await this.mailer.sendMemberInviteMail(target.email, {
```

Comment:
`mailer.sendMemberInviteMail()` should belong to notification service biz logic.